app/src/main/java/bes/max/playlistmaker/data/player/PlayerImpl.kt

```diff
@@ -1,64 +1,55 @@
 package bes.max.playlistmaker.data.player
 
-import android.media.AudioAttributes
 import android.media.MediaPlayer
-import bes.max.playlistmaker.domain.player.Player
+import android.util.Log
 import bes.max.playlistmaker.domain.models.PlayerState
+import bes.max.playlistmaker.domain.player.Player
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 
-class PlayerImpl : Player {
+private const val TAG = "PlayerImpl"
+
+class PlayerImpl(private val mediaPlayer: MediaPlayer) : Player {
 
-    private var mediaPlayer: MediaPlayer? = null
     private val _playerState = MutableStateFlow<PlayerState>(PlayerState.STATE_DEFAULT)
     override val playerState: StateFlow<PlayerState> = _playerState.asStateFlow()
 
-    init {
-        createNewMediaPlayer()
-    }
-
-    private fun createNewMediaPlayer() {
-        mediaPlayer = MediaPlayer().apply {
-            setAudioAttributes(
-                AudioAttributes.Builder()
-                    .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
-                    .setUsage(AudioAttributes.USAGE_MEDIA)
-                    .build()
-            )
-        }
-    }
-
     override fun preparePlayer(dataSourceUrl: String) {
-        if (mediaPlayer != null) {
-            with(mediaPlayer!!) {
-                setDataSource(dataSourceUrl)
-                prepareAsync()
-                setOnPreparedListener { _playerState.value = PlayerState.STATE_PREPARED }
-                setOnCompletionListener { _playerState.value = PlayerState.STATE_PREPARED }
-            }
+        with(mediaPlayer!!) {
```

Comment:
🍏 Теперь mediaPlayer ненуллабельный и можно убрать double bang:
```suggestion
        with(mediaPlayer) {
```